test/jdk/javax/swing/JButton/HtmlButtonImageTest/HtmlButtonImageTest.java

```diff
@@ -0,0 +1,158 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+*/
+
+/*
+ * @test
+ * @key headful
+ * @bug 8015854
+ * @requires (os.family == "mac")
+ * @summary Tests HTML image as JButton text for unwanted padding on Aqua LAF on MacOS
+ * @run main HtmlButtonImageTest
+ */
+
+import java.awt.Dimension;
+import java.awt.FlowLayout;
+import java.awt.Point;
+import java.awt.Robot;
+import java.nio.file.Path;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import java.awt.Color;
+import java.io.IOException;
+import javax.imageio.ImageIO;
+import java.io.File;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+
+public final class HtmlButtonImageTest {
+    private static JFrame frame;
+    private static Point point;
+    private static JButton button;
+    private static Path testDir;
+
+    public static final int BUTTON_HEIGHT = 37;
+    public static final int BUTTON_WIDTH = 37;
+    public static final int SQUARE_HEIGHT = 19;
+    public static final int SQUARE_WIDTH = 19;
+    public static final int PIXEL_BUFFER = 1;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            UIManager.setLookAndFeel("com.apple.laf.AquaLookAndFeel");
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException("Unsupported LookAndFeel: " + e);
+        }
+
+        Robot robot = new Robot();
+        robot.setAutoDelay(2000);
+        robot.setAutoWaitForIdle(true);
+
+        // store path to source directory to locate image
+        testDir = Path.of(System.getProperty("test.classes", "."));
+
+        // generate red_square.png image to use as JButton text
+        generateImage();
+
+        SwingUtilities.invokeAndWait(() -> {
+            createAndShowGUI();
+        });
+
+        // retrieve color of pixels at each edge of square image by starting at the center of the button
+        robot.mouseMove(frame.getLocationOnScreen().x, frame.getLocationOnScreen().y);
+        robot.mouseMove(button.getLocationOnScreen().x, button.getLocationOnScreen().y);
+
+        setupCenterCoord();
+        robot.mouseMove(point.x, point.y);
+
+        // store each pixel color on the edge of each side of the red square
+        Color leftClr = robot.getPixelColor(point.x - (SQUARE_WIDTH/2) + PIXEL_BUFFER, point.y);
+        Color rightClr = robot.getPixelColor(point.x + (SQUARE_WIDTH/2) - PIXEL_BUFFER, point.y);
+        Color topClr = robot.getPixelColor(point.x, point.y - (SQUARE_HEIGHT/2) + PIXEL_BUFFER);
+        Color botClr = robot.getPixelColor(point.x, point.y + (SQUARE_HEIGHT/2) - PIXEL_BUFFER);
```

Comment:
there was an issue of using getPixelColor on retina screen on mac laptop as @honkar-jdk  found out in #[7219](https://git.openjdk.java.net/jdk/pull/7219)..She has to use BufferedImage..Are you sure this is working on all mac CI system, mac external monitor and mac laptop?

Reply:
Initially, I ran into this issue as well a few weeks ago when I made the test. My solution was to check for redness by checking for 250 or greater R values with 10 or less G and B values instead of checking for exactly 255,0,0. When tested in mach5, some Macs failed with values that had pixels that fit within this color range, but wasn't exactly 255,0,0.